budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -1,247 +1,1041 @@
+        for i in range(10):
+            if i < 5:
+                province = provinces[0]
             else:
-                status = self.statuses[1]
-                province = self.provinces[1]
-                source = self.sources[1]
-
-            ProvInfraProject.objects.create(
-                financial_year=self.fin_year,
-                IRM_project_id=i,
+                province = provinces[1]
+            project = ProvInfraProject.objects.create(IRM_project_id=i)
+            ProvInfraProjectSnapshot.objects.create(
+                irm_snapshot=self.irm_snapshot,
+                project=project,
                 name="Project {}".format(i),
-                department="Department {}".format(i),
-                local_municipality="Local {}".format(i),
-                district_municipality="District {}".format(i),
                 province=province,
-                status=status,
-                primary_funding_source=source,
-                main_contractor="Contractor {}".format(i),
-                principle_agent="Principle Agent {}".format(i),
-                program_implementing_agent="Program Agent {}".format(i),
-                other_parties="Service Provider: XXXX{}".format(i),
+                estimated_completion_date=date(year=2020, month=1, day=1),
             )
+        ProvInfraProjectIndex().reindex()
 
-    def test_projects_per_page(self):
-        response = self.client.get(self.url, format="json")
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    def tearDown(self):
+        ProvInfraProjectIndex().clear()
+        self.file.close()
+
+    def test_search_homepage_correct_numbers(self):
+        selenium = self.selenium
+        selenium.get("%s%s" % (self.live_server_url, self.url))
+        num_of_projects = selenium.find_element_by_css_selector(
+            "#num-matching-projects-field"
+        ).text
+        num_of_projects = int(num_of_projects)
+        self.assertEqual(num_of_projects, 11)
+
+        num_of_items = len(
+            selenium.find_elements_by_css_selector(
+                "#result-list-container .narrow-card_wrapper"
+            )
+        )
+        self.assertEqual(num_of_projects, num_of_items)
+
+    def test_number_updated_after_search(self):
+        province = "Eastern Cape"
+        selenium = self.selenium
+        selenium.get("%s%s" % (self.live_server_url, self.url))
+        num_of_projects = selenium.find_element_by_css_selector(
+            "#num-matching-projects-field"
+        ).text
+        num_of_projects = int(num_of_projects)
+        self.assertEqual(num_of_projects, 11)
+
+        search_field = selenium.find_element_by_css_selector(
+            "#Infrastructure-Search-Input"
+        )
+        search_button = selenium.find_element_by_css_selector("#Search-Button")
+        search_field.send_keys(province)
+        search_button.click()
+        wait = WebDriverWait(selenium, self.timeout)
+        wait.until(
```

Comment:
it doesn't help to do wait.until for something that is already on the page.

The point of using wait.until is so that selenium waits until something that should be updated by javascript has been updated.

In this case, you are waiting until a heading which is built into the html is on the page after you clicked a button - that button won't reload the page, so it's immediately successful - so this doesn't end up waiting for the asynchronous javascrip stuff to complete.